### DISCUSSION TITLE
Account management

### DISCUSSION THREAD

**Discussion by Jocky92** [pglombardo/PasswordPusher#1041] [Q&A] [answered]
We would like to host PasswordPusher ourselves in the future with all the given features as well as the file upload.
In terms of privacy, we do not want to allow accounts to be created by people outside the company. The file upload should only be possible by people of our company. Is there a possibility to manage the accounts or to make such restrictions?

**Comment by Jocky92** [comment]
im sorry didnt saw this post. duplicate of [Control who can create an account #356](https://github.com/pglombardo/PasswordPusher/discussions/356)
or is there any news about the thmea since then?

**Reply by pglombardo** [reply]
That post is out of date - I'll update.

**Comment by pglombardo** [comment, marked as answer]
To control accounts, [enable logins](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#enabling-logins) and set `PWP__ALLOW_ANONYMOUS` to false.  This blocks all anonymous users from creating text, file or URL pushes.

Then once all of your team mate accounts are made, set `PWP__DISABLE_SIGNUPS` which will disable new account registration.

Anonymous users will still be able to receive/retrieve secret URLs that are given to them.

In the future, I plan to add extended support for teams but that hasn't happened yet.

Details in the [Configuration.md document](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#enabling-logins).

__August 2023 Update__: We now support limiting sign ups to specific email domains.  See `PWP__SIGNUP_EMAIL_REGEXP` in the [Configuration document](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#enabling-logins).

Let me know if this helps.

**Reply by Jocky92** [reply]
@pglombardo thank you for your explanation. The procedure would be useful if we had to create only a few accounts. With about 150 users this is a bit difficult. 
In addition, I ask myself how I can ensure that the account is deleted when the person leaves the company again. 
A connection to the AD would be nice here. Otherwise I need a kind of SuperAdmin, with which it is possible to perform such operations as creating and deleting users.

**Reply by pglombardo** [reply]
I'm investigating adding SSO currently.  As for AD integration, what do you use currently?

**Reply by Jocky92** [reply]
I am not sure if I understand the question correctly. We currently have Kerberos, LDAP, SSO and SAML in use. But if I see it correctly, there is no PasswordPusher support for any of them yet.

**Reply by pglombardo** [reply]
That helps thanks - I'm trying to map out the best path for Password Pusher to support the most login options for the user base without adding too much complexity.  Then also something that will work on pwpush.com and self-hosted.  There is a lot to it.

**Reply by pglombardo** [reply]
Hi @Jocky92 - slightly relevant, we now support limiting signups by email domain with `PWP__SIGNUP_EMAIL_REGEXP`.  I updated the original answer.  I hope this improves the situation a bit.

**Reply by Jocky92** [reply]
Hi @pglombardo 
Thank you for adding this feature!! This definitely helps us a bit.
Now if it would be possible to generate an output about all existing accounts and delete/deactivate single ones then this would be perfect for our scenario. Deleting/deactivating an account would still be important in case a person leaves the company. Whether via GUI or CLI would not matter for us.

**Reply by pglombardo** [reply]
I agree - the app badly needs support for teams and administration of the team members.  Then I'd like to also add team policies to enforce rules for generated pushes.  Unfortunately, I only do this in my free time so I appreciate the patience.  But it's happening albeit slowly.
